src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -1281,6 +1258,26 @@ class RideHailManager(
     }
   }
 
+  private def getActualResponses(request: RideHailRequest) = {
+    val previousResponses = rideHailResponseCache.getOrElse(request.customer, IndexedSeq.empty)
+    val currentTime = request.requestTime.getOrElse(0)
+    previousResponses.filter(_.request.departAt >= currentTime)
+  }
+
+  private def removeOriginalResponseFromCache(request: RideHailRequest) = {
+    val actualResponses: IndexedSeq[RideHailResponse] = getActualResponses(request)
+    val departureTimeOrdering =
+      Ordering.by((rsp: RideHailResponse) => Math.abs(request.departAt - rsp.request.departAt))
```

Comment:
Is `Math.abs` the correct option? This will potentially result in lower values being made into higher values (i.e. request's depart = 5 and depart responses of 3 and 6 you get 2 and 1, respectively - so the higher, 6 is chosen). Plus, see my comment on the reduction, if it is needed or not

Reply:
We need to choose the closest by time (and maybe origin and destination locations) original response. Generally, a person during mode choice makes multiple RH requests (direct RH trip, and 2 RH legs for a RideHailTansit trip). Then the person chooses a trip and makes a Reservation request that is matched to the original response here. It's not necessarily the earliest one.
In your example, 6 should be chosen.